lib/palindrome_check.rb

```diff
@@ -1,5 +1,39 @@
+require 'pry'
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
+
+#try regex gsub(/\w+/)
+#gsub returns a copy of string with all occurences of patter for the given argument.
+# w+ - Matches any word character
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+
+  if my_phrase.nil?
+    return false
+  end
+
+  if my_phrase == ""
+    return true
+  end
+
+  # Get rid of white space if white space exists
+  if my_phrase.gsub!(/\W/, "") != nil
```

Comment:
Can you think of a way to account for white spaces without altering the input string?